src/chapters/classes/constructors.rst

```diff
@@ -0,0 +1,157 @@
+Constructors
+============
+
+We’ll often want to initialize, or set the initial value of, some of our
+fields when creating a new object from a class. **Constructors** allow us to do so.
+
+In Java, constructors have the same name as the class and are most often
+declared public (though they can have any other valid access modifier).
+They are declared *without a return type*. Any function that is named
+the same as the class and has no return type is a constructor.
+
+Here is an example of a constructor definition within the ``HelloWorld``
+class:
+
+.. sourcecode:: java
+   :linenos:
+
+   public class HelloWorld {
+
+       private String message = "Hello World";
+
+       public HelloWorld(String message) {
+           this.message = message;
+       }
+
+       public String getMessage() {
+           return message;
+       }
+
+       public void setMessage(String aMessage) {
+           message = aMessage;
+       }
+
+   }
+
+This constructor allows us to create ``HelloWorld`` objects with custom
+messages. The assignment ``this.message = message`` assigns the value
+passed into the constructor to the field ``message``. Here’s how we
+might use it:
+
+.. sourcecode:: java
+
+   HelloWorld goodbye = new HelloWorld("Goodbye World");
+   System.out.println(goodbye.getMessage()); // prints "Goodbye World"
+
+.. note:: 
+
+   It’s not required that every class have a constructor. If you don’t
+   provide one, the Java compiler will generate an “empty” constructor for
+   you, known as a **default constructor**. For example, when we left out a
```

Comment:
since default constructor is an index entry candidate, maybe this content is better suited for the main test rather than a note admonition?

Reply:
Done! Thanks!